core/types/calls.cc

```diff
@@ -995,12 +1002,42 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
                     continue;
                 }
                 consumed.insert(spec.name);
-                TypeAndOrigins tpe;
-                tpe.origins = {kwargsLoc};
+                // We want the precise location of the kwarg that we are processing.
+                // To do that, we rely on hash->{keys,values} being added in the order
+                // that the kwargs appear in the the call.  We also rely on args from
+                // a shapely-typed kwsplat hash (implicit or explicit) being added to
+                // hash->{keys,values} after the explicitly-named kwargs in the call.
                 auto offset = arg - hash->keys.begin();
+                // offset * 2 skips kwargs we have already seen.
+                // + 1 moves us to the value being passed.
+                ENFORCE(!sawKwSplat);
+                auto kwaValue = kwait + (offset * 2) + 1;
```

Comment:
```suggestion
                auto kwaValue = kwait + nonPosArgs;
```
The number of non-positional arguments is named right before positional arguments are processed.

Reply:
`kwait` points to the beginning of the keyword arguments (it's not modified at all during this loop), so `kwait + nonPosArgs` would always point at the same thing, namely the argument just after all the keyword arguments, which I don't think is what we want?

Reply:
Ah, for some reason I totally read past the fact that it's using `offset` in the calculation of `kwaValue`! You can ignore this suggestion :)